daemon/logger/copier.go

```diff
@@ -44,7 +49,14 @@ func (c *Copier) Run() {
 
 func (c *Copier) copySrc(name string, src io.Reader) {
 	defer c.copyJobs.Done()
-	buf := make([]byte, bufSize)
+
+	var buf []byte
+	if sizedLogger, ok := c.dst.(SizedLogger); ok {
+		buf = make([]byte, sizedLogger.BufSize())
+	} else {
+		buf = make([]byte, defaultBufSize)
```

Comment:
I think it would be more readable to make this:

```Go
bufSize := defaultBufSize
if sizedLogger, ok := c.dst.(SizedLogger); ok {
  bufSize = sizedLogger.BufSize()
}
buf := make([]byte, bufSize) 
```

Reply:
I like it. I'll change and test (with the other requested changes).